src/Components/WebAssembly/WebAssembly/src/Rendering/WebAssemblyRenderer.cs

```diff
@@ -187,16 +187,16 @@ private async Task ProcessNextDeferredEventAsync()
         readonly struct IncomingEventInfo
         {
             public readonly ulong EventHandlerId;
-            public readonly EventFieldInfo EventFieldInfo;
+            public readonly EventFieldInfo? EventFieldInfo;
             public readonly EventArgs EventArgs;
-            public readonly TaskCompletionSource<object> TaskCompletionSource;
+            public readonly TaskCompletionSource TaskCompletionSource;
 
-            public IncomingEventInfo(ulong eventHandlerId, EventFieldInfo eventFieldInfo, EventArgs eventArgs)
+            public IncomingEventInfo(ulong eventHandlerId, EventFieldInfo? eventFieldInfo, EventArgs eventArgs)
             {
                 EventHandlerId = eventHandlerId;
                 EventFieldInfo = eventFieldInfo;
                 EventArgs = eventArgs;
-                TaskCompletionSource = new TaskCompletionSource<object>();
+                TaskCompletionSource = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
Behavior change.

Reply:
@pranavkm I missed this before. Can you explain why this needs to be changed? Why is the `RunContinuationsAsynchronously` behavior the right one here? Under what circumstances might this break people?